### ISSUE TITLE
More bindings and Urho update

### ISSUE THREAD

**Issue by Alan-FGR** [xamarin/urho#296]
Hello there.
I was implementing binary serialization but when I was halfway into it I realized the AbstractFile (which inherits both Serializer and Deserializer) and superclasses aren't bound yet, on the native side they provide type-specific write/read methods that are piped into a single method for that purpose. Since they are many methods I wonder if you guys are planning to update Urho soon and perhaps you could also point the binding generator at those classes required for binary serialization, they're not complex classes but of course there's the problem of multiple inheritance. Worst case scenario I guess it could be reimplemented fully managed by passing pointer and size in the callback.

**Comment by Alan-FGR** [comment]
Also, lemme take this opportunity to ask another question: I see that you need to add the callbacks in Node.cpp, are you guys using some specific program to add those there or is it fully manual (and rebased/merged directly with git+some diff merger)?

**Comment by Alan-FGR** [comment]
By the way, on a side note, luckily Urho creates a VectorBuffer for each Component when serializing what makes it pretty easy and safe to use for managed components.

**Comment by EgorBo** [comment]
@Alan-FGR yes, the callbacks to managed side in Node are hand written so you can add yours if you need them :-)
I'll try to bind AbstractFile

**Comment by Alan-FGR** [comment]
@EgorBo yes, no problem in that. The only thing really missing is the binding since manually doing that is too much work, the serialization step itself I can do. It would be very helpful if that class could be bound :). Thank you.
P.S.: any plans to update UrhoSharp to a newer Urho version? :P

**Comment by EgorBo** [comment]
@Alan-FGR after I push what I have now to nuget - it usually takes less than an hour to update all bindings and test everything (unless urho has some SDL updates)

**Comment by Alan-FGR** [comment]
@EgorBo no hurry. Thank you.

**Comment by troll-kadabra** [comment]
Hi. Is this done?

**Comment by troll-kadabra** [comment]
Hi. Any new?

**Comment by troll-kadabra** [comment]
not news on this?
